lib/errors.c

```diff
@@ -1107,32 +1107,38 @@ int nterror_to_errno(uint32_t status) {
         case SMB2_STATUS_CANNOT_DELETE:
         case SMB2_STATUS_FILE_DELETED:
                 return EPERM;
-        case SMB2_STATUS_NO_MORE_FILES:
+		case SMB2_STATUS_NO_MORE_FILES:
                 return ENODATA;
         case SMB2_STATUS_LOGON_FAILURE:
                 return ECONNREFUSED;
-        case SMB2_STATUS_NOT_A_DIRECTORY:
+		case SMB2_STATUS_NOT_A_DIRECTORY:
                 return ENOTDIR;
         case SMB2_STATUS_NOT_IMPLEMENTED:
         case SMB2_STATUS_INVALID_DEVICE_REQUEST:
+
         case SMB2_STATUS_ILLEGAL_FUNCTION:
-        case SMB2_STATUS_INVALID_PARAMETER:
+
+		case SMB2_STATUS_INVALID_PARAMETER:
         case SMB2_STATUS_NOT_SUPPORTED:
         case SMB2_STATUS_NOT_A_REPARSE_POINT:
                 return EINVAL;
+#ifndef XBOX_PLATFORM
```

Comment:
Instead of adding #ifdefs here in this file,
can we just define SMB2_STATUS_STOPPED_ON_SYMLINK to be some random value in the xbox compatibility header files?

That is usually how I deal with "define is missing on platform foo", I add a dummy #define foo whatever to the compatibility files.
This is all to avoid having platform specific #ifdefs int he main code.

Reply:
Solved.